File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/comment/direct/Java/Python/s196249385_commented.py
def main():
    n = int(input())
    k = int(input())
    a = list(map(int, input().split()))
    
    cumsum = [0] * (n + 1)
    for i in range(n):
        cumsum[i+1] = cumsum[i] + a[i]
    
    b = []
    max_sum = 0
    for i in range(n):
        for j in range(i, n):
            current = cumsum[j+1] - cumsum[i]
            b.append(current)
            if current > max_sum:
                max_sum = current
    
    # Compute the highest bit
    if max_sum == 0:
        mask = 
    else:
        bit_length = max_sum.bit_length()
        mask = 1 << (bit_length - 1)
    
    current_mask = mask
    while current_mask > 0:
        tmp = []
        for e in b:
            if (e & current_mask) != 0:
                tmp.append(e)
        if len(tmp) >= k:
            b = tmp
        current_mask >>= 1
    
    if len(b) < k:
        print(0)
    else:
        ans = -1
        for e in b:
            ans &= e
        print(ans)
    
if __name__ == "__main__":
    main()
# End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/comment/direct/Java/C++/s196249385_commented.cpp
#include <iostream>
#include <vector>
using namespace std;

void solve() {
    int n, k;
    cin >> n >> k;
    vector<int> a(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }

    vector<long> cumsum(n + 1);
    cumsum[0] = 0;
    for (int i = 0; i < n; i++) {
        cumsum[i + 1] = cumsum[i] + a[i];
    }

    vector<long> b;
    long max_val = 0;
    for (int i = 0; i < n; i++) {
        for (int j = i; j < n; j++) {
            long s = cumsum[j + 1] - cumsum[i];
            b.push_back(s);
            if (s > max_val) {
                max_val = s;
            }
        }
    }

    long mask = 0;
    if (max_val !=  pute 0) {
        mask = 1;
        while (mask <= max_val) {
            mask <<= 1;
        }
        mask >>= 1;
    }

    while (mask > 0) {
        vector<long> tmp;
        for (auto e : b) {
            if ((e & mask) != 0) {
                tmp.push_back(e);
            }
        }
        if (tmp.size() >= k) {
            b = tmp;
        }
        mask >>= 1;
    }

    if (b.size() < k) {
        cout << 0 << endl;
    } else {
        long ans = -1;
        for (auto e : b) {
            ans &= e;
        }
        cout << ans << endl;
    }
}

int main() {
    solve();
    return 0;
}
// End of Code
